pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,221 @@
+        try {
+            cmd = parseCommandLineArgs(options, args);
+        } catch (ParseException e) {
+            System.out.format("%s.%n", e.getMessage());
+            final HelpFormatter formatter = new HelpFormatter();
+            formatter.printHelp("HelloWorldReader", options);
+            System.exit(1);
+        }
+
+        final String scope = cmd.getOptionValue("scope") == null ? Constants.DEFAULT_SCOPE : cmd.getOptionValue("scope");
+        final String stream = cmd.getOptionValue("stream") == null ? Constants.DEFAULT_STREAM_NAME : cmd.getOptionValue("stream");
+
+        final String uriString = cmd.getOptionValue("uri") == null ? Constants.DEFAULT_CONTROLLER_URI : cmd.getOptionValue("uri");
+        final URI controllerURI = URI.create(uriString);
+
+        final String truststorePath =
+                cmd.getOptionValue("truststore") == null ? Constants.DEFAULT_TRUSTSTORE_PATH : cmd.getOptionValue("truststore");
+        final boolean validateHostname = cmd.getOptionValue("validatehost") == null ? false : true;
+
+        final String username = cmd.getOptionValue("username") == null ? Constants.DEFAULT_USERNAME : cmd.getOptionValue("username");
+        final String password = cmd.getOptionValue("password") == null ? Constants.DEFAULT_PASSWORD : cmd.getOptionValue("password");
+
+        SecureBatchReader reader = new SecureBatchReader(scope, stream, controllerURI,
+                truststorePath, validateHostname,
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    private int readFromSegments(BatchClientFactory batchClient, List<SegmentRange> segments) {
+        List<Integer> batchEventCountList = new ArrayList<>();
+        UTF8StringSerializer serializer = new UTF8StringSerializer();
```

Comment:
Are the other samples using this serializer, right? Otherwise it will have problems to read data from other samples.